independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/MultipartMessageBodyWriter.java

```diff
@@ -153,7 +153,7 @@ private String getFileNameIfFile(Object value, String partFileName) {
         } else if (value instanceof FileDownload) {
             return "; filename=\"" + ((FileDownload) value).fileName() + "\"";
         } else if (partFileName != null) {
```

Comment:
Should we move this condition to the top, so users can override file name for File{,Download} as well?

Reply:
I think the filename sets in the FileDownload should be used instead of the one in the part item. 
Though I'm not sure how RESTEasy classic behaves in this case.
Do you know @geoand ?

Reply:
I do not

Reply:
I'll leave it for now then.